src/IncomingWebhook.ts

```diff
@@ -38,15 +38,16 @@ export class IncomingWebhook {
    */
   public async send(message: string): Promise<IncomingWebhookResult | undefined> {
     // NOTE: no support for TLS config
-     let  payload = message;
-
+    let  payload = message;
 
     try {
       const response = await this.axios.post(this.url, payload);
 
       return this.buildResult(response);
     } catch (error) {
-            throw error;
+      // make troubleshooting API failures a bit easier by pinning the response body on the error
+      error.text = this.buildResult(error.response).text;
```

Comment:
> I think we can shorten this code to `error.response.data`

I chose to use `error.text` to make this similar to the returned `response.text` value of the 'happy flow'

>  I don't quite understand why you put the error response data back into the error object.

This value already exists as `error.response.data`, as you mentioned. But it doesn't get logged / serialized with the error object, because the `error.response` object relates to the HTTP request made by `axios`. So pinning it separately as `error.text` allows it to be logged with the error.

> I think it makes more sense if we return our own error object / interface with a data attribute.

Yes, I like this idea a lot! It will make the error reporting of this client independent of the HTTP client it uses. I'll redo my PR to create a standalone `Error` object in case of failure, and populate it with relevant bits of data from the `axios` error object, such as the HTTP status code and response body.

Did I capture your feedback properly?

Reply:
Yes, I have just saw it. Then it definitely makes sense to add the error object under error.text again. 
> Did I capture your feedback properly?
yeah, that sounds awesome.